source/server/options_impl.cc

```diff
@@ -179,6 +179,13 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv,
     concurrency_ = std::max(1U, concurrency.getValue());
   }
 
+  // Exactly one of config_path and config_yaml should be specified.
+  if (!config_path.isSet() && !config_yaml.isSet()) {
+    const std::string message =
```

Comment:
no need to have this variable?

Reply:
This particular check (verifying that either config_path or config_yaml was provided on the command line) proved to be problematic, causing all sorts of tests to fail with the exception thrown here. I'm open to suggestions on how to remediate this, but I don't want to start breaking assumptions about how Envoy is started with this PR.

Reply:
I mean you could save the message variable and simply raise with a literal string.

Reply:
This section has been removed. I changed the equivalent error message in `source/server/server.cc` to use a string literal.